app/models/product_import/entry_processor.rb

```diff
@@ -60,45 +63,38 @@ def count_existing_items
     end
 
     def reset_absent_items
-      # For selected enterprises; set stock to zero for all products/inventory
-      # that were not listed in the newly uploaded spreadsheet
-      return unless data_for_stock_reset?
-      suppliers_to_reset_products = []
-      suppliers_to_reset_inventories = []
+      return unless settings.data_for_stock_reset? && settings.reset_all_absent?
 
-      settings = @import_settings[:settings]
+      reset_absent.call
```

Comment:
I am not a big fan of the delegate line above:
`delegate :products_reset_count, to: :reset_absent`

I'd prefer a similar but more explicit solution like this:
on top:
`     - attr_reader :products_reset_count`
on the initialize method
`     - @products_reset_count = 0`
and on this line:
`     - @products_reset_count = reset_absent.call`

I think this solution is less smart... which is better.